include/engine/geospatial_query.hpp

```diff
@@ -47,6 +47,42 @@ template <typename RTreeT, typename DataFacadeT> class GeospatialQuery
         return rtree.SearchInBox(bbox);
     }
 
+    std::vector<PhantomNodeWithDistance>
+    NearestPhantomNodes(const util::Coordinate input_coordinate,
```

Comment:
I know it's an internal API, but might be worth changing `NearestPhantomNodes` with optional `max_results` parameter to be a required parameter, to clarify the distinction between the two functions.

Reply:
Out of interest, did you evaluate this approach for when `max_results` is set also?

Reply:
>I know it's an internal API, but might be worth changing NearestPhantomNodes with optional max_results parameter to be a required parameter, to clarify the distinction between the two functions.

Great idea, updated :+1:

> Out of interest, did you evaluate this approach for when max_results is set also?

Nope, but I thought about it: we probably could make another optimisation when both `max_results` and `max_distance` are set. Or may be you see how it can be applied when only `max_results` is set?